java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/AutoOtelAgentRunConfigurationWrapper.kt

```diff
@@ -53,50 +48,17 @@ class AutoOtelAgentRunConfigurationExtension : RunConfigurationExtension() {
     run configuration with a gradle script generated in place and a JavaExec task that runs the main method.
     unit tests will be executed by calling gradle :test task.
      */
-
-
-    override fun isApplicableFor(configuration: RunConfigurationBase<*>): Boolean {
-        Log.log(logger::debug, "isApplicableFor, project:{}, id:{}, name:{}, type:{}",
-                configuration.project, configuration.id, configuration.name, configuration.type)
-
-        val runConfigType = evalRunConfigType(configuration)
-
-        return runConfigType != RunConfigType.Unknown
-    }
-
-
     override fun <T : RunConfigurationBase<*>?> updateJavaParameters(
         configuration: T & Any,
         params: JavaParameters,
         runnerSettings: RunnerSettings?,
+        resolvedModule: Module?,
     ) {
 
-        val resolvedModule = RunCfgTools.resolveModule(configuration, params, runnerSettings)
-
-        Log.log(logger::debug, "updateJavaParameters, project:{}, id:{}, name:{}, type:{}, module:{}",
-            configuration.project, configuration.id, configuration.name, configuration.type, resolvedModule)
-
         val project = configuration.project
-        val runConfigType = evalRunConfigType(configuration)
-        val autoInstrumentationEnabled = enabled()
-        val connectedToBackend = isBackendOk(project)
-
-        reportToPosthog(project, runConfigType, autoInstrumentationEnabled, connectedToBackend)
-
-        //testing if enabled must be done here just before running.
-        if (!autoInstrumentationEnabled) {
-            Log.log(logger::debug, "AutoOtelAgentRunConfigurationExtension is not enabled")
-            return
-        }
-
-        if (!connectedToBackend) {
-            Log.log(logger::warn, "No connection to Digma backend. going to skip the Otel agent")
-            return
-        }
-
         val useOtelAgent = evalToUseAgent(resolvedModule)
 
-        when (runConfigType) {
+        when (evalRunConfigType(configuration)) {
```

Comment:
for easier debug please use this pattern
```suggestion
        val runConfigType = evalRunConfigType(configuration)
        when (runConfigType) {
```